### PR TITLE
PT-526 | Event wizard step 2

### PR DESCRIPTION
**Pull request by Sartonon**
## Description

- Add second step to event wizard (occurrences)

### Closes
**[PT-526](https://helsinkisolutionoffice.atlassian.net/browse/PT-526):**

### CODE REVIEW

File: src/domain/occurrence/CreateOccurrencePage.tsx [hunk 1]
```diff
@@ -179,4 +229,33 @@ const CreateOccurrencePage: React.FC = () => {
...
+            endsAt: initialEndsAt,
+          }
+        : defaultInitialValues;
+    }
+    return defaultInitialValues;
+  }, [initialDate, initialEndsAt, initialStartsAt]);
+
+  return initialFormValues;
+};
+
```
Comment (Sartonon) [inline src/domain/occurrence/CreateOccurrencePage.tsx:260]: wasn't quite sure what would be the cleanest way to handle these initial values... ended up with this :D

File: src/domain/event/CreateEventPage.tsx [hunk 2]
```diff
@@ -119,7 +122,14 @@ const CreateEventPage: React.FC = () => {
...
       history.push({
         pathname: `/${locale}${ROUTES.CREATE_OCCURRENCE.replace(':id', id)}`,
+        search: firstOccurrencePrefilledValuesToQuery(values),
       });
+
+      // TODO: think if there is better way to handle this scrollTop.
+      // at the moment <ScrollToTop /> component ignores
+      // '/events/:id/occurrences/create' because it matches one of the ignored paths:
+      // '/events/:id/occurrences/:occurrenceId'
+      window.scrollTo(0, 0);
```
Comment (jorilindell) [inline src/domain/event/CreateEventPage.tsx:132]: Could you add new prop to ScrollToTop where you define array of routes that should alway scroll to top. In that case you could /events/:id/occurrences/:occurrenceId to ignored routes and /events/:id/occurrences/create to the routes (I don't know what could be suitable name for that prop) that should always scroll to top

File: src/domain/occurrence/CreateOccurrencePage.tsx [hunk 3]
```diff
@@ -179,4 +229,33 @@ const CreateOccurrencePage: React.FC = () => {
   );
 };
 
+const useInitialFormValues = () => {
```
Comment (jorilindell) [inline src/domain/occurrence/CreateOccurrencePage.tsx:232]: Is this same hook also needed on edit page? If yes, you could move this to separate file

File: src/domain/occurrence/EditOccurrencePage.tsx [hunk 4]
```diff
@@ -152,6 +167,19 @@ const EditOccurrencePage: React.FC = () => {
...
+  const handleDeleteOccurrence = async (
+    occurrence: OccurrenceFieldsFragment
+  ) => {
+    try {
+      await deleteOccurrence({ variables: { input: { id: occurrence.id } } });
+      refetchEvent();
+    } catch (e) {
+      toast(t('occurrences.deleteError'), {
+        type: toast.TYPE.ERROR,
+      });
```
Comment (jorilindell) [inline src/domain/occurrence/EditOccurrencePage.tsx:179]: You could also write this line as
```
toast.error(t('occurrences.deleteError'));
```
In that case could can test it easier with Jest (If you want to test the error case). e.g. `toast.error = jest.fn()`

File: src/hooks/useQuery.ts [hunk 5]
```diff
@@ -1,5 +1,12 @@
...
 export const useQuery = () => {
-  return new URLSearchParams(useLocation().search);
+  const search = useLocation().search;
+
+  const query = React.useMemo(() => {
+    return new URLSearchParams(search);
+  }, [search]);
+
+  return query;
 };
```
Comment (jorilindell) [inline src/hooks/useQuery.ts:12]: Maybe you could rename this hook e.g. useSearchParams. At first time when I read this PR through I wasn't sure what it does

**Review by jorilindell — APPROVED** [review]
Looks good. I added couple of comments but they are not deal breakers to me

File: src/domain/occurrence/CreateOccurrencePage.tsx (hunk 1 above)
Comment (Sartonon) [reply src/domain/occurrence/CreateOccurrencePage.tsx:260]: did some changes to this already for the next step

File: src/domain/occurrence/CreateOccurrencePage.tsx (hunk 3 above)
Comment (Sartonon) [reply src/domain/occurrence/CreateOccurrencePage.tsx:232]: at least not yet, I will move it if it is needed. But for now only create occurrence page needs initialValues from query params

Comment (jorilindell) [reply src/domain/occurrence/CreateOccurrencePage.tsx:232]: Ok, in that case there is no need to do anything for this

File: src/hooks/useQuery.ts (hunk 5 above)
Comment (Sartonon) [reply src/hooks/useQuery.ts]: renamed

File: src/domain/occurrence/EditOccurrencePage.tsx (hunk 4 above)
Comment (Sartonon) [reply src/domain/occurrence/EditOccurrencePage.tsx]: fixed

File: src/domain/event/CreateEventPage.tsx (hunk 2 above)
Comment (Sartonon) [reply src/domain/event/CreateEventPage.tsx]: added new prop to ScrollToTop component and use that to handle the scrolling

File: src/domain/occurrence/CreateOccurrencePage.tsx [hunk 6]
```diff
@@ -135,16 +158,31 @@ const CreateOccurrencePage: React.FC = () => {
...
+  const handleDeleteOccurrence = async (
+    occurrence: OccurrenceFieldsFragment
+  ) => {
+    try {
+      await deleteOccurrence({ variables: { input: { id: occurrence.id } } });
+      refetchEvent();
+    } catch (e) {
+      toast(t('occurrences.deleteError'), {
+        type: toast.TYPE.ERROR,
+      });
```
Comment (jorilindell) [inline src/domain/occurrence/CreateOccurrencePage.tsx:170]: Sorry, I missed this at first time. Maybe could use `toast.error(t('occurrences.deleteError'))` also here

File: src/domain/app/ScrollToTop.tsx [hunk 7]
```diff
@@ -4,27 +4,39 @@ import { matchPath, useLocation } from 'react-router';
...
+        })
+      ),
+    [locale, pathname]
+  );
 
-    if (!isIgnoredPath) {
-      window.scrollTo(0, 0);
+  React.useEffect(() => {
+    if (isMatch(forceScrollToTopPaths)) {
+      debugger;
```
Comment (jorilindell) [inline src/domain/app/ScrollToTop.tsx:28]: Is this line still needed?

**Review by jorilindell — APPROVED** [review]
Looks good. Added some really minor comments

File: src/domain/app/ScrollToTop.tsx (hunk 7 above)
Comment (Sartonon) [reply src/domain/app/ScrollToTop.tsx:28]: nope :D

**Pull request merged**